hack/verify-shellcheck.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck source=/dev/null
+ROOT_PATH=$(git rev-parse --show-toplevel)
+
+# create a temporary directory
+TMP_DIR=$(mktemp -d)
+
+# cleanup on exit
+cleanup() {
+  echo "Cleaning up..."
+  rm -rf "${TMP_DIR}"
+}
+trap cleanup EXIT
+
+# check for shellcheck installation
+SHELLCHECK_VERSION=0.6.0
+HAVE_SHELLCHECK=false
+if which shellcheck &>/dev/null; then
+  detected_version="$(shellcheck --version | grep 'version: .*')"
+  if [[ "${detected_version}" = "version: ${SHELLCHECK_VERSION}" ]]; then
+    HAVE_SHELLCHECK=true
+  fi
+fi
+if ! ${HAVE_SHELLCHECK}; then
+  # install shellcheck (Linux-x64 only!)
+  if [[ "$OSTYPE" != "darwin"* ]]; then
+    cd "${TMP_DIR}" || exit
+    VERSION="shellcheck-v0.6.0"
+    DOWNLOAD_FILE="${VERSION}.linux.x86_64.tar.xz"
+    wget https://storage.googleapis.com/shellcheck/"${DOWNLOAD_FILE}"
+    tar xf "${DOWNLOAD_FILE}"
+    cd "${VERSION}" || exit
```

Comment:
nit: what is this for? do we need this?

Reply:
I found that that it just checks that the tar file expanded into a directory properly, and that you can `cd` into it (the new directory is named `shellcheck-v0.6.0`. I can remove this if you like, but I can see some value having it.